megamek/src/megamek/common/Entity.java

```diff
@@ -2287,23 +2300,34 @@ public boolean isElevationValid(int assumedElevation, IHex hex) {
         int assumedAlt = assumedElevation + hex.surface();
         if (getMovementMode() == EntityMovementMode.VTOL) {
             if ((this instanceof Infantry)
-                && (hex.containsTerrain(Terrains.BUILDING)
-                    || hex.containsTerrain(Terrains.WOODS) || hex
-                    .containsTerrain(Terrains.JUNGLE))) {
+                    && (hex.containsTerrain(Terrains.BUILDING)
+                            || hex.containsTerrain(Terrains.WOODS) 
+                            || hex.containsTerrain(Terrains.JUNGLE))) {
                 // VTOL BA (sylph) can move as ground unit as well
                 return ((assumedElevation <= 50) && (assumedAlt >= hex.floor()));
-            } else if (hex.containsTerrain(Terrains.BRIDGE_ELEV)) {
-                // fly under a bridge as long as there is enough clearance below and above the unit
-                return (assumedElevation <= 50)
-                        && assumedElevation > hex.floor()
-                        && (assumedElevation > hex.terrainLevel(Terrains.BRIDGE_ELEV)
-                                || assumedElevation + height() < hex.terrainLevel(Terrains.BRIDGE_ELEV) - 1);
-            } else if (hex.containsTerrain(Terrains.WOODS)
-                       || hex.containsTerrain(Terrains.WATER)
-                       || hex.containsTerrain(Terrains.JUNGLE)) {
-                return ((assumedElevation <= 50) && (assumedAlt > hex.ceiling()));
-            }
-            return ((assumedElevation <= 50) && (assumedAlt >= hex.ceiling()));
+            } else {
+                // VTOLs can be anywhere on or above ground and fly beneath bridges,
+                // land on buildings and ignore planted fields and industrial zone 
+                // but cannot land on water or trees
+                // As always, height() reports one less than the rules (TW p.99) say
+                // Units may move under a bridge if their top is 
+                // lower than or equal to the bridge height (TW p.62)
+                boolean allowed = (assumedElevation <= 50) && (assumedElevation >= 0);
+                if (hex.containsTerrain(Terrains.BRIDGE_ELEV)) {
+                    allowed &= (assumedElevation >= hex.terrainLevel(Terrains.BRIDGE_ELEV))
+                            || (assumedElevation + height() + 1 <= hex.terrainLevel(Terrains.BRIDGE_ELEV));
```

Comment:
Is `<=` proper for "under" a bridge?

Reply:
TW says a unit may move under a bridge so long as the height of the unit is less than or equal to the bridge level. I.E. a mech fits under a level 2 bridge. A heli, elevation 1 and height 1 (0 in MM for whatever reason), should also fit under a bridge. I think (hope) this one is correct.